domain-model.md

```diff
@@ -0,0 +1,46 @@
+# Basket
+
+<!---
+1.
+As a member of the public,
+So I can order a bagel before work,
+I'd like to add a specific type of bagel to my basket.
+----------------------------------------------------------------------------------------------
+2.
+As a member of the public,
+So I can change my order,
+I'd like to remove a bagel from my basket.
+----------------------------------------------------------------------------------------------
+3.
+As a member of the public,
+So that I can not overfill my small bagel basket
+I'd like to know when my basket is full when I try adding an item beyond my basket capacity.
+----------------------------------------------------------------------------------------------
+4.
+As a Bob's Bagels manager,
+So that I can expand my business,
+I’d like to change the capacity of baskets.
+----------------------------------------------------------------------------------------------
+5.
+As a member of the public
+So that I can maintain my sanity
+I'd like to know if I try to remove an item that doesn't exist in my basket.
+-->
+
+| Classes | Methods                               | Scenario                              | Outputs |
+|---------|---------------------------------------|---------------------------------------|---------|
+| Basket  | add(String bagelname)                 | if the bagel has been added           | true    |
+|         |                                       | if the bagel has not been found       | false   |
+|         |                                       |                                       |         |
+|         | remove(String name)                   | if the bagel is still in the list     | false   |
+|         |                                       | if the bagel has been removed         | true    |
+|         |                                       |                                       |         |
+|         | capacityCheck()                       | if max capacity has been exceeded     | false   |
```

Comment:
I would probably rename the `capacityCheck()` method to something that makes the return values more obvious, so that when you read the method call in code, the true/false return value is easier to reason about. If you called it `spaceAvailable()` or something similar then when it returns `true` it is easy to understand the value you get back.